xarray/tests/test_backends.py

```diff
@@ -1704,6 +1704,90 @@ def test_raise_on_forward_slashes_in_names(self) -> None:
                 with self.roundtrip(ds):
                     pass
 
+    @requires_netCDF4
+    def test_encoding_enum__no_fill_value(self):
+        with create_tmp_file() as tmp_file:
+            cloud_type_dict = {"clear": 0, "cloudy": 1}
+            with nc4.Dataset(tmp_file, mode="w") as nc:
+                nc.createDimension("time", size=2)
+                cloud_type = nc.createEnumType("u1", "cloud_type", cloud_type_dict)
+                v = nc.createVariable(
+                    "clouds",
+                    cloud_type,
+                    "time",
+                    fill_value=None,
+                )
+                v[:] = 1
+            with open_dataset(tmp_file) as ds:
+                assert (
+                    ds.clouds.encoding["dtype"].metadata["enum_dict"] == cloud_type_dict
+                )
+                assert ds.clouds.encoding["dtype"].metadata["enum_name"] == "cloud_type"
+                with create_tmp_file() as tmp_file2:
+                    ds.to_netcdf(tmp_file2)
```

Comment:
Can we test that this survives a roundtrip? I think we need to adapt the encoding workflow to allow this.

Reply:
Indeed.

Reply:
@bzah I've tinkered a bit with this today. I can provide more details tomorrow. I could even push a fix to your branch then, if that's OK with you.

Reply:
Yes of course, I will just push the comments fixes now.
I suspect the encoding["dtype"] is edited somewhere when encode_cf is called but I haven't found where yet.